### PR TITLE
Cosmo: make example dev files operational

### PR DESCRIPTION
**Pull request by nstarman**
The example Cosmology I/O files are now fully operational and pytested!

Closes: https://github.com/astropy/astropy/projects/7#card-67535480
Suggested by @mhvk 

Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/cosmology/mypackage/io/astropy_convert.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    # ----------------
+    # remap metadata
+    m["t0"] = m.pop("current_age") * u.Gyr
+
+    # optional
+    if "reionization_redshift" in m:
+        m["z_reion"] = m.pop("reionization_redshift")
+
+    ...  # keep building `m`
```
Comment (mhvk) [inline docs/cosmology/mypackage/io/astropy_convert.py:67]: I guess this works - an empty `Ellipsis` does nothing...  But maybe just as a comment nonetheless?

File: docs/cosmology/mypackage/io/astropy_convert.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    meta = m.pop("meta")
+    m = {**meta, **m}  # merge, preferring current values
+
+    # ----------------
+    # remap values
+    # MyCosmology doesn't support units, so take values.
+    m["hubble_parameter"] = m.pop("H0").to_value(u.km/u.s/u.Mpc)
+    m["initial_matter_density"] = m.pop("Om0")
+    m["initial_temperature"] = m.pop("Tcmb0").to_value(u.K)
+    m["Neff"] = m.pop("Neff")
```
Comment (mhvk) [inline docs/cosmology/mypackage/io/astropy_convert.py:114]: I would not change "Neff" here - maybe have a comment explicitly stating that you can skip it?

File: docs/cosmology/mypackage/io/astropy_convert.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    """
+    m = dict(mycosmo)
+    m["name"] = mycosmo.name
+
+    # ----------------
+    # remap Parameters
+    m["H0"] = m.pop("hubble_parameter") * (u.km / u.s / u.Mpc)
+    m["Om0"] = m.pop("initial_matter_density")
+    m["Tcmb0"] = m.pop("initial_temperature") * u.K
+    m["Neff"] = m.pop("Neff")
```
Comment (mhvk) [inline docs/cosmology/mypackage/io/astropy_convert.py:55]: Like below, perhaps not change things that already are OK?

**Review by mhvk — APPROVED** [review]
This is great! I really only have nitpicks - though admittedly I didn't look super carefully. But I think carefully enough to approve, especially as this is meant as example code that people should adjust.

File: docs/cosmology/mypackage/io/astropy_convert.py (hunk 1 above)
Comment (nstarman) [reply docs/cosmology/mypackage/io/astropy_convert.py]: SGTM.
Done!

File: docs/cosmology/mypackage/io/astropy_convert.py (hunk 2 above)
Comment (nstarman) [reply docs/cosmology/mypackage/io/astropy_convert.py]: SGTM.
Done!

File: docs/cosmology/mypackage/io/astropy_convert.py (hunk 3 above)
Comment (nstarman) [reply docs/cosmology/mypackage/io/astropy_convert.py]: SGTM.
Done!

**Comment by nstarman** [comment]
> This is great! I really only have nitpicks - though admittedly I didn't look super carefully. But I think carefully enough to approve, especially as this is meant as example code that people should adjust.

Thanks! I think I'm almost ready to approach 3rd party packages. Let there be interoperability.

**Pull request merged**